### DISCUSSION TITLE
Message Header Properties

### DISCUSSION THREAD

**Discussion by ChristophBachmann** [Havret/dotnet-activemq-artemis-client#436] [Ideas]
In Java, there is the possibility using the JSMContext (Java Message Service) to add specific header properties to a message (using setProperty(KEY, VALUE) method). Here is a code example in Java:

````java
final var jmsProducer = jmsContext.createProducer();
    final String traceId = UUID.randomUUID().toString();
    jmsProducer.setProperty(TRACE_ID, traceId);
   // set another properties e.g. VERSION, ORIGIN, CONTENT_TYPE, CREATED_AT
````

Is there a possibility to set those properties in Artemis Message as well? Would it be possible to add this feature, or maybe another workaround? It would be very nice and helpful to be able to send those message header properties.

Best regards
Chris

**Comment by Havret** [comment]
Hello @ChristophBachmann, 

The API you're looking for is available via message `Message.ApplicationProperties`. 

For instance 
```csharp
var message = new Message("my-payload");
const string TRACE_ID = "TRACE_ID";
string traceId = Guid.NewGuid().ToString("D");
message.ApplicationProperties[TRACE_ID] = traceId;

var producer = await connection.CreateProducerAsync("a1", RoutingType.Anycast);
await producer.SendAsync(message);
```

Best,
Havret

**Reply by ChristophBachmann** [reply]
Hello @Havret 

You are awesome! Thank you very much - it worked 👍 

Best,
Chris
